lib/src/functions/color.dart

```diff
@@ -447,112 +420,86 @@ final _scale = _function("scale", r"$color, $kwargs...", (arguments) {
   }
 
   var keywords = Map.of(argumentList.keywords);
-  num getScale(String name) {
-    var value = keywords.remove(name);
-    if (value == null) return null;
-    var number = value.assertNumber(name);
-    number.assertUnit("%", name);
-    return number.valueInRange(-100, 100, name) / 100;
-  }
 
-  num scaleValue(num current, num scale, num max) {
-    if (scale == null) return current;
-    return current + (scale > 0 ? max - current : current) * scale;
+  /// Gets and validates the parameter with [name] from keywords.
+  ///
+  /// [max] should be 255 for RGB channels, 1 for the alpha channel, and 100
+  /// for saturation, lightness, whiteness, and blackness.
+  num getParam(String name, num max) {
+    var number = keywords.remove(name)?.assertNumber(name);
+    if (number == null) return null;
+    if (scale) {
+      number.assertUnit("%", name);
+      return number.valueInRange(-100, 100, name) / 100;
+    }
+    return number.valueInRange(adjust ? -max : 0, max, name);
   }
 
-  var red = getScale("red");
-  var green = getScale("green");
-  var blue = getScale("blue");
-  var saturation = getScale("saturation");
-  var lightness = getScale("lightness");
-  var alpha = getScale("alpha");
+  var alpha = getParam("alpha", 1);
+  var red = getParam("red", 255);
+  var green = getParam("green", 255);
+  var blue = getParam("blue", 255);
+  var hue = scale ? null : keywords.remove("hue")?.assertNumber("hue")?.value;
+  var saturation = getParam("saturation", 100);
+  var lightness = getParam("lightness", 100);
+  var whiteness = getParam("whiteness", 100);
+  var blackness = getParam("blackness", 100);
 
```

Comment:
This should assert that these parameters (but not `saturation` and `lightness`) have unit `%` for `adjust()` and `change()`.